main/src/com/google/refine/commands/GetAllPreferencesCommand.java

```diff
@@ -52,9 +52,9 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
             throws ServletException, IOException {
         
         Project project = request.getParameter("project") != null ? getProject(request) : null;
-        PreferenceStore ps = project != null ? 
-                project.getMetadata().getPreferenceStore() : 
-                ProjectManager.singleton.getPreferenceStore();
+        
+        // project level has no specific preference
+        PreferenceStore ps = ProjectManager.singleton.getPreferenceStore();
                 
         try {
             response.setCharacterEncoding("UTF-8");
```

Comment:
I'm not sure I understand this change - do we used to have project-level preferences that have been removed by this PR?

Reply:
No. nothing is changed. but just adding some comments to make it clear

Reply:
well you did change the code too - `project.getMetadata().getPreferenceStore()` is not called anymore, so it's not just about adding some comments.

Reply:
it's remove and added backed as above

Reply:
no @jackyq2015 it is not "added backed as above", there is a difference in the code and I would like you to explain it to me. I accept to turn a blind eye on some dirty corners of this PR but it would be nice if you could cooperate a bit more… I am trying to help you improve your PR.

Reply:
Let me paste the diff and explain:
-                ProjectManager.singleton.getPreferenceStore();      // 1 line was removed 
+        
+        // project level has no specific preference                       // 1 comment added
+        PreferenceStore ps = ProjectManager.singleton.getPreferenceStore();  // 1 line was added back

Reply:
Well then we are not seeing the same diffs!

This is what I see:

Three lines removed:
```
-        PreferenceStore ps = project != null ? 
-                project.getMetadata().getPreferenceStore() : 
-                ProjectManager.singleton.getPreferenceStore();
```

Three lines added:
```
+        
+        // project level has no specific preference
+        PreferenceStore ps = ProjectManager.singleton.getPreferenceStore();
```

So do we agree that there is a change in the code? You removed a condition. It does not seem harmful to me, but that is what I wanted to discuss… Makes sense?

Reply:
It's actually a change worth being discussed, because it means that the project passed as parameter to the command is now completely ignored - so that means the preceding line can be removed too:
`Project project = request.getParameter("project") != null ? getProject(request) : null;`

But that is suspicious - why was there any need to pass this argument before? Why would the preference store be different depending on the project? Do we rely on that anywhere? If so, there is a risk that this PR introduces a bug. I am not saying it does, I am just saying this needs to be investigated.

Reply:
The diff given by github didn't provided 3 "-" for some reason. it only shows 1 "-".

Reverted the change

Reply:
Great, thanks a lot! (But that's worrying - it means we can't rely on GitHub to review PRs?! That sounds like a serious bug to me!)